### ISSUE TITLE
System.ArgumentException

### ISSUE THREAD

**Issue by muekoeff** [naseweis520/osu-Sync#23]
[osu!Forum](https://osu.ppy.sh/forum/p/4772055)

``` Indent
=====   osu!Sync Crash | 2015-12-28 12:02:13   =====

// Information
An exception occured in osu!Sync. If this problem persists please report it using the Feedback-window, on GitHub or on the osu!Forum.
When reporting please try to describe as detailed as possible what you've done and how the applicationen reacted.
GitHub: http://j.mp/1PDuDFp   |   osu!Forum: http://j.mp/1PDuCkK

// Configuration
{"application":{"isElevated":"False","lastUpdateCheck":"28-12-2015 11:43:04","version":"1.0.0.7"},"config":{"downloadMirror":"2","updateInterval":"3"},"language":{"code":{"long":"en_US","short":"en"}},"system":{"cultureInfo":"en-GB","is64bit":"True","operatingSystem":"6.3.9600.0"}}

// Exception
System.ArgumentException: Illegal characters in path.
   at System.IO.Path.CheckInvalidPathChars(String path, Boolean checkAdditional)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.File.InternalReadAllBytes(String path, Boolean checkHost)
   at osu_Sync.MainWindow.Importer_Downloader_DownloadFileCompleted(Object sender, AsyncCompletedEventArgs e)
   at System.Net.WebClient.OnDownloadFileCompleted(AsyncCompletedEventArgs e)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by muekoeff** [comment]
Reported by [oroplox](https://osu.ppy.sh/forum/p/4800002):

_when it comes to that 32k maps pack _[[Ref](https://osu.ppy.sh/forum/t/399688)]_... it seems like you only get the error when using osu.uu.gl mirror while bloodcat works just fine (tho it's way slower). The error is triggered by specific maps and you can "bypass" it by manually downloading the map that caused it, you can see the ID of the map that caused the crash at the bottom where it says "downloading X" before closing the error message
(2 crashes i remember had \* in maps name // 260539 Raspberry*Monster and 103659 Monochrome*HAPPY DAY)_

Seems to be because of an invalid file name.

**Comment by muekoeff** [comment]
[Beatmap List with troublesome beatmaps.](https://github.com/naseweis520/osu-Sync/files/83566/Issue.23.zip)

**Comment by muekoeff** [comment]
Fixed with 82e1d5cf0950ef65968504b3a4fd693ca7f1d030
